system/jlib/jmetrics.hpp

```diff
@@ -262,4 +273,42 @@ std::shared_ptr<CustomMetric<T>> createCustomMetricAndAddToReporter(const char *
     return pMetric;
 }
 
+
+class jlib_decl ScopedGaugeUpdater
+{
+public:
+    ScopedGaugeUpdater(const std::shared_ptr<GaugeMetric> &_pGauge, int64_t _amount=1):
+        pGauge{_pGauge},
+        amount{_amount}
+    {
+        pGauge->inc(amount);
+    }
+
+    ~ScopedGaugeUpdater()
+    {
+        pGauge->dec(amount);
+    }
+
+protected:
+    std::shared_ptr<GaugeMetric> pGauge;
+    int64_t amount;
+};
+
+
+class jlib_decl AutoGaugeDecrement
+{
+public:
+    AutoGaugeDecrement(const std::shared_ptr<GaugeMetric> &_pGauge, int64_t _amount=1):
+        pGauge{_pGauge},
+        amount{_amount} { }
+    ~AutoGaugeDecrement()
+    {
+        pGauge->dec(amount);
+    }
+
+protected:
+    std::shared_ptr<GaugeMetric> pGauge;
```

Comment:
as per comment above, can this be a GaugeMetric or a reference to a shared_ptr to avoid the overhead?